ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKTable.java

```diff
@@ -318,6 +319,52 @@ public SchemaKGroupedTable groupBy(
     );
   }
 
+  public <KRightT> SchemaKTable<K> foreignKeyJoin(
```

Comment:
nit meta: for function names, I'd suggest we use `XXInnerJoinYY` instead of `XXJoinYY` for all such cases since `JOIN` is just a syntax sugar for `INNER JOIN` here, and the function name would better be with the full name to clearly indicate their scope.